docs/catalog-restrictions.md

```diff
@@ -0,0 +1,194 @@
+spec:
+  authInfo:
+    basic:
+      secretRef:
+        name: sample-broker-auth
+        namespace: brokers
+  catalogRestrictions:
+    servicePlan:
+    - "spec.externalName==basic"
+  url: http://sample-broker.brokers.svc.cluster.local
+```
+
+In this example, a catalog restriction has been defined that specifies that 
+only service plans that have an external name of basic should be selected. 
+Catalog restrictions are defined as a set of one or more rules that target 
+either service classes and service plans. These rules have a special format
+ similar to Kubernetes label selectors. 
+
+The rule format is expected to be `<property><conditional><requirement>`
+
+* `<property>` is one of the supported properties of a service class or service plan resource, described below
+* `<conditional>` is allowed to be one of the following: `==`, `!=`, `in`, `notin`
+* `<requirement>` will be a string value if `==` or `!=` are used, otherwise it will be a set of string values if `in` or `notin` are used
+* `<requirement>` is case sensitive
+
+Catalog restrictions, while similar to label selectors, only operate on a 
+subset of properties on service class and service plan resources. The following
+ sections detail what properties can be used to define catalog restrictions for
+  each resource type.  
+
+`ClusterServiceClass` allowed property names:
+
+| Property Key    | Description    |
+| name |  This key will match the ClusterServiceClass.Name property |
+| spec.externalName | This key will match the ClusterServiceClass.Spec.ExternalName property |
+| spec.externalID | This key will match the ClusterServiceClass.Spec.ExternalID property |
+
+`ServiceClass` allowed property names:
+
+| Property Key    | Description    |
+| name |  This key will match the ServiceClass.Name |
+| spec.externalName | This key will match the ServiceClass.Spec.ExternalName property |
+| spec.externalID | This key will match the ServiceClass.Spec.ExternalID property |
+
+`ClusterServicePlan` allowed property names:
+
+| Property Key    | Description    |
+| name | This key will match the ClusterServicePlan.Name |
+| spec.externalName | This key will match the ClusterServicePlan.Spec.ExternalName property |
+| spec.externalID | This key will match the ClusterServicePlan.Spec.ExternalID property |
+| spec.free | This key will match the ClusterServicePlan.Spec.Free property |
+| spec.clusterServiceClass.name | This key will match the ClusterServicePlan.Spec.ClusterServiceClassRef.Name property |
+
+`ServicePlan` allowed property names:
+
+| Property Key    | Description    |
+| name | This key will match the ServicePlan.Name property |
+| spec.externalName | This key will match the ServicePlan.Spec.ExternalName property |
+| spec.externalID | This key will match the ServicePlan.Spec.ExternalID property |
+| spec.free | This key will match the ServicePlan.Spec.Free property |
+| spec.serviceClass.name | This key will match the ServicePlan.Spec.ServiceClassRef.Name property |
+
```

Comment:
Since the property key and descriptions are the same for the NS and Cluster scoped Plan/Class, I'd suggest combining them.  IE  "`ClusterServicePlan` and `ServicePlan` allowed property names"
Same with the Classes

Reply:
They are _almost_ the same. There are some shared keys, but we also allow to supplies predicates that reference the non-common parts, i.e. ServicePlans can be filtered with ServiceClassRef.Name and ClusterServicePlans can be filtered with ClusterServiceClassRef.Name.

Would you favor doing a shared block and then call out the two differences?

Reply:
@jeremyrickard I missed that.  I totally agree, leave it as is, thank you.